### DISCUSSION TITLE
Taking user input in obsidian

### DISCUSSION THREAD

**Discussion by whoisteve** [Taitava/obsidian-shellcommands#323] [Q&A]
When writing a shell script, it may be helpful to prompt the user for input. This will allow you to get a specific value that you can use in your script as you see fit. It also allows you to make your script interactive by prompting the user to enter values such as directories and filenames. 

When it comes to an user input, obsidian can open an pop-up where the user can type the text.

When it comes to an interactive action, the command aborts.

```sh
read my_var
```

**Comment by Taitava** [comment]
Hi, thank you for your suggestion! 🙂 It's already possible to [take input from a user by using Prompts](https://publish.obsidian.md/shellcommands/Preactions/Prompts).

However, it currently supports just singleline text fields. There is a list of [planned input types](https://github.com/Taitava/obsidian-shellcommands/discussions/17#discussion-3576274) (see the _Input types_ heading) that I might implement some day. If you come up with more ideas regarding prompts, please go ahead and comment in the linked discussion - I'm open for feedback. 🙂 

(I'll move this discussion from _Ideas_ to _Q & A_).
